ec/resources/waf/rate_rule.go

```diff
@@ -283,19 +251,57 @@ func ResourceRateRuleRead(
 	d.Set("name", resp.Name)
 	d.Set("num", resp.Num)
 	d.Set("keys", resp.Keys)
-	flattenedConditionGroups := FlattenConditionGroups(resp.ConditionGroups)
+	flattenedConditionGroups := flattenConditionGroups(resp.ConditionGroups)
 	d.Set("condition_group", flattenedConditionGroups)
 	return diags
 }
 
-func ResourceRateRuleUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+func ResourceRateRuleUpdate(
+	ctx context.Context,
+	d *schema.ResourceData,
+	m interface{},
+) diag.Diagnostics {
+	config := m.(**api.ClientConfig)
+	wafService, err := buildWAFService(**config)
 
-	var diags diag.Diagnostics
-	return diags
-}
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	ruleID := d.Id()
+	rule, err := expandRateRule(d)
+	if err != nil {
+		d.SetId("")
```

Comment:
Since this is the update case, do we want to set an empty ID if there is an error generated here? I believe this would be a case where the user makes a mistake in the terraform configuration file. Since the resource already exists, should we just return the error?

Reply:
Correct, missed that. Thanks for catching it.